### PR TITLE
delete ineffectual assignment

### PR DESCRIPTION
**Pull request by yu8833**
Signed-off-by: yupengzte <yu.peng36@zte.com.cn>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by vdemeester — COMMENTED** [review]
Interesting
/cc @dperny why don't we check for errors here ?

**Comment by dperny** [comment]
Something is seriously wrong with this code. I think there may have been a bad merge or rebase, where it merged "cleanly" but not really. Like, I know I didn't call `TaskLogs` twice, and if I would have, that never would have made it through review. 

I don't know what has happened, but this isn't the correct fix.

File: cli/command/service/logs.go
Comment (dperny) [inline cli/command/service/logs.go]: Instead of removing the err, do this:

```go
task, _, err := cli.TaskInspectWithRaw(ctx, opts.target)
if err != nil {
    if client.IsErrTaskNotFound(err) {
        // if the task isn't found, rewrite the error to be clear
        // that we looked for services AND tasks and found none
        err = fmt.Errorf("No such task or service")
    }
    return err
}
```

Basically, moving lines 106-113 to this point in the code.

File: cli/command/service/logs.go
Comment (dperny) [inline cli/command/service/logs.go]: After you move the above code to the other noted spot, add a simple error check here for the result of `cli.TaskLogs`

```go
responseBody, err := cli.TaskLogs(ctx, opts.target, options)
if err != nil {
    return err
}
```

File: cli/command/service/logs.go
Comment (dperny) [inline cli/command/service/logs.go]: Finally, delete this second call to `cli.TaskLogs` entirely. I have no idea how this got here. We shouldn't be calling this twice.

**Review by dperny — CHANGES_REQUESTED** [review]

**Comment by dperny** [comment]
Oh my god that totally did make it through code review 🙃 💀

File: cli/command/service/logs.go
Comment (YuPengZTE) [inline cli/command/service/logs.go]: `responseBody, err := cli.TaskLogs(ctx, opts.target, options)`
Here, is `:=` OK? It cause `responseBody declared and not used`, it is better like
responseBody, err = cli.TaskLogs(ctx, opts.target, options)

**Comment by YuPengZTE** [comment]
The second fix has done. 
@dperny PTAL.

File: cli/command/service/logs.go
Comment (dperny) [inline cli/command/service/logs.go]: Yes, you are correct. about this.

**Comment by dperny** [comment]
I am not a maintainer, but LGTM, assuming tests pass. Good catch @YuPengZTE. I really dropped the ball on this code and you've pointed out and fixed some really glaring mistakes.

File: cli/command/service/logs.go
Comment (duglin) [inline cli/command/service/logs.go]: can you include the thing we're looking for in the error message itself?  It helps users track down their bugs faster.

**Comment by duglin** [comment]
one minor request, otherwise LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 🦁

File: cli/command/service/logs.go
Comment (dperny) [inline cli/command/service/logs.go]: It doesn't *really* work to specify whether we're looking for a service or a task, because we first check for service and then we check for task, and there isn't really a way to differentiate the two based on the user's input.

Comment (duglin) [inline cli/command/service/logs.go]: can't we just print `opts.target` ?

Comment (dperny) [inline cli/command/service/logs.go]: AH that's what you meant. So, like

```go
err = fmt.Errorf("no such task or service: %v", opts.target")
```

Comment (duglin) [inline cli/command/service/logs.go]: yep - sorry if I wasn't clear before

**Comment by tonistiigi** [comment]
LGTM

**Pull request merged**
